This is a patch to mosfet's liquid 0.7.
Features:
 - Qt-only
 - works with Qt/E on QPE/OPIE
 
--- -	Mon Jun 24 19:05:49 2002
+++ liquid.h	Fri Jun 21 14:48:21 2002
@@ -2,7 +2,7 @@
 #define LIQUID_STYLE_H
 
 
-#include <kstyle.h>
+#include <qwindowsstyle.h>
 #include <qpainter.h>
 #include <qdrawutil.h>
 #include <qpalette.h>
@@ -20,7 +20,7 @@
  * Copyright (C)QPL 2001 Daniel M. Duley. All rights reserved.
  */
 
-class KPixmap;
+class QPixmap;
 
 #define BITMAP_ITEMS 41
 #define LIQUID_MENU_CHANGE 667
@@ -50,24 +50,22 @@
     ~TransMenuHandler(){;}
     void reloadSettings();
     int transType(){return(type);}
-    KPixmap *pixmap(WId id){return(pixDict.find(id));}
+    QPixmap *pixmap(WId id){return(pixDict.find(id));}
     const QColor& textColor(){return(fgColor);}
     const QColor& bgColor(){return(color);}
     bool useShadowText(){return(shadowText);}
-protected slots:
-    void slotKIPCMessage(int id, int arg);
 protected:
     bool eventFilter(QObject *obj, QEvent *ev);
-    void stripePixmap(KPixmap &pix, const QColor &color);
+    void stripePixmap(QPixmap &pix, const QColor &color);
 
     QColor color, fgColor;
     int opacity;
     int type;
     bool shadowText;
-    QIntDict<KPixmap>pixDict;
+    QIntDict<QPixmap>pixDict;
 };
 
-class LiquidStyle : public KStyle
+class LiquidStyle : public QWindowsStyle
 {
     friend class TransMenuHandler;
 public:
@@ -121,33 +119,14 @@
                     bool tickAbove, bool tickBelow);
     void drawSliderMask(QPainter *p, int x, int y, int w, int h,
                         Orientation orient, bool, bool);
-    void drawKToolBar(QPainter *p, int x, int y, int w, int h,
-                      const QColorGroup &g, KToolBarPos type,
-                      QBrush *fill=NULL);
-    void drawKBarHandle(QPainter *p, int x, int y, int w, int h,
-                        const QColorGroup &g,
-                        KToolBarPos type, QBrush *fill=NULL);
-    void drawKMenuBar(QPainter *p, int x, int y, int w, int h,
-                      const QColorGroup &g, bool macMode,
-                      QBrush *fill=NULL);
-    void drawKToolBarButton(QPainter *p, int x, int y, int w, int h,
-                            const QColorGroup &g, bool sunken=false,
-                            bool raised = true, bool enabled = true,
-                            bool popup = false,
-                            KToolButtonType icontext = Icon,
-                            const QString& btext=QString::null,
-                            const QPixmap *icon=NULL,
-                            QFont *font=NULL, QWidget *btn=NULL);
-    void drawKMenuItem(QPainter *p, int x, int y, int w, int h,
-                       const QColorGroup &g, bool active,
-                       QMenuItem *item, QBrush *fill=NULL);
+	void drawMenuBarItem(QPainter *p, int x, int y, int w, int h,
+                            QMenuItem *mi, QColorGroup &g, bool enabled, bool active );
+
     void drawPopupMenuItem(QPainter *p, bool checkable, int maxpmw,
                            int tab, QMenuItem *mi, const QPalette &pal,
                            bool act, bool enabled, int x, int y, int w,
                            int h);
     int popupMenuItemHeight(bool c, QMenuItem *mi, const QFontMetrics &fm);
-    void drawKProgressBlock(QPainter *p, int x, int y, int w, int h,
-                            const QColorGroup &g, QBrush *fill);
     void drawFocusRect(QPainter *p, const QRect &r, const QColorGroup &g,
                        const QColor *pen, bool atBorder);
     int defaultFrameWidth() const {return(2);}
@@ -167,12 +146,6 @@
     void drawToolButton(QPainter *p, int x, int y, int w,
                         int h, const QColorGroup &g,
                         bool sunken, const QBrush *fill);
-    void drawKickerAppletHandle(QPainter *p, int x, int y, int w, int h,
-                                const QColorGroup &g, QBrush *);
-    void drawKickerTaskButton(QPainter *p, int x, int y, int w, int h,
-                              const QColorGroup &g,
-                              const QString &title, bool active,
-                              QPixmap *icon, QBrush *fill);
     // for repainting toolbuttons when the toolbar is resized
     bool eventFilter(QObject *obj, QEvent *ev);
     void drawSliderGroove(QPainter * p, int x, int y, int w,  int h,
--- -	Mon Jun 24 19:05:49 2002
+++ liquid.cpp	Fri Jun 21 22:02:55 2002
@@ -10,12 +10,9 @@
 
 #include <qmenudata.h>
 #include "liquid.h"
-#include <kapp.h>
-#include <kglobal.h>
-#include <kconfig.h>
-#include <kdrawutil.h>
-#include <kglobalsettings.h>
-#include <kpixmapeffect.h>
+#include <qapplication.h>
+#include <qpe/config.h>
+#include "effects.h"
 #include <qpalette.h>
 #include <qbitmap.h>
 #include <qtabbar.h>
@@ -25,28 +22,29 @@
 #include <qtimer.h>
 #include <qpixmapcache.h>
 #include <qradiobutton.h>
-#include <kimageeffect.h>
-#include <ktoolbar.h>
+#include <qcombobox.h>
 #include <qdrawutil.h>
 #include <qwidgetlist.h>
 #include <qtoolbutton.h>
 #include <qheader.h>
 #include <unistd.h>
-#include <klocale.h>
-#include <kiconloader.h>
-#include <kmenubar.h>
-#include <kipc.h>
+#include <qmenubar.h>
 
-#include <X11/X.h>
-#include <X11/Xlib.h>
+
+#include <stdio.h>
 
 #include "htmlmasks.h"
 #include "embeddata.h"
 
-void TransMenuHandler::stripePixmap(KPixmap &pix, const QColor &color)
+typedef void (QStyle::*QDrawMenuBarItemImpl) (QPainter *, int, int, int, int, QMenuItem *,
+                          QColorGroup &, bool, bool);
+
+QDrawMenuBarItemImpl qt_set_draw_menu_bar_impl(QDrawMenuBarItemImpl impl);
+
+void TransMenuHandler::stripePixmap(QPixmap &pix, const QColor &color)
 {
     QImage img(pix.convertToImage());
-    KImageEffect::fade(img, 0.9, color);
+    QImageEffect::fade(img, 0.9, color);
     int x, y;
     int r, g, b;
     for(y=0; y < img.height(); y+=3){
@@ -71,24 +69,21 @@
     : QObject(parent)
 {
     pixDict.setAutoDelete(true);
-    connect(kapp, SIGNAL(kipcMessage(int, int)), this,
-            SLOT(slotKIPCMessage(int, int)));
     reloadSettings();
 }
 
 void TransMenuHandler::reloadSettings()
 {
     pixDict.clear();
-    KConfig *config = KGlobal::config();
-    config->setGroup("MosfetMenus");
 
-    type = config->readNumEntry("Type", TransStippleBg);
-    color = config->readColorEntry("Color",
-                                   &QApplication::palette().active().button());
-    fgColor = config->readColorEntry("TextColor",
-                                     &QApplication::palette().active().text());
-    opacity = config->readNumEntry("Opacity", 10);
-    shadowText = config->readBoolEntry("ShadowText", true);
+    Config config ( "qpe" );
+    config. setGroup ( "MosfetMenus" );
+
+    type = config. readNumEntry("Type", TransStippleBg);
+    color = QColor ( config. readEntry("Color",  QApplication::palette().active().button().name()));
+    fgColor = QColor ( config. readEntry("TextColor", QApplication::palette().active().text().name()));
+    opacity = config. readNumEntry("Opacity", 10);
+    shadowText = config. readBoolEntry("ShadowText", true);
 }
 
 bool TransMenuHandler::eventFilter(QObject *obj, QEvent *ev)
@@ -99,7 +94,7 @@
         if(type == TransStippleBg || type == TransStippleBtn ||
            type == Custom){
             QApplication::syncX();
-            KPixmap *pix = new KPixmap;
+            QPixmap *pix = new QPixmap;
             if(p->testWFlags(Qt::WType_Popup)){
                 QRect r(p->x(), p->y(), p->width(), p->height());
                 QRect deskR = QApplication::desktop()->rect();
@@ -107,7 +102,7 @@
                     r.setBottom(deskR.bottom());
                     r.setRight(deskR.right());
                 }
-                *pix = QPixmap::grabWindow(qt_xrootwin(), r.x(), r.y(),
+                *pix = QPixmap::grabWindow(QApplication::desktop()-> winId(), r.x(), r.y(),
                                            r.width(), r.height());
             }
             else{ // tear off menu
@@ -121,7 +116,7 @@
                 stripePixmap(*pix, p->colorGroup().button());
             }
             else{
-                KPixmapEffect::fade(*pix, (((float)opacity)+80)*0.01, color);
+                QPixmapEffect::fade(*pix, (((float)opacity)+80)*0.01, color);
             }
             pixDict.insert(p->winId(), pix);
         }
@@ -129,74 +124,19 @@
     else if(ev->type() == QEvent::Hide){
         if(type == TransStippleBg || type == TransStippleBtn ||
            type == Custom){
-            qWarning("Deleting menu pixmap, width %d", pixDict.find(p->winId())->width());
+//            qWarning("Deleting menu pixmap, width %d", pixDict.find(p->winId())->width());
             pixDict.remove(p->winId());
         }
     }
     return(false);
 }
 
-void TransMenuHandler::slotKIPCMessage(int id, int)
-{
-    if(id == LIQUID_MENU_CHANGE){
-        bool oldShadow = shadowText;
-
-        KConfig *config = KGlobal::config();
-        config->reparseConfiguration(); // in case KControl changed values
-
-        // Make sure no popupmenus are shown. There shouldn't be any because
-        // the user just clicked "Apply", but there can be tear offs ;-)
-        // We just close them so the pixmaps are deleted and regenerated.
-        QWidgetList *list = QApplication::topLevelWidgets();
-        QWidgetListIt it( *list );
-        QWidget *w;
-        while ((w=it.current()) != 0 ){
-            ++it;
-            if(w->inherits("QPopupMenu")){
-                w->close();
-            }
-        }
-
-        reloadSettings();
-
-        // Now repaint menubar if needed
-        if(shadowText != oldShadow){
-            it.toFirst();
-            while ((w=it.current()) != 0 ){
-                ++it;
-                if(w->inherits("QMenuBar")){
-                    w->repaint();
-                }
-            }
-        }
-    }
-    else if(id == MOSFET_BUTTON_CHANGE){
-        qWarning("In mosfet button change");
-        // really, this should be in LiquidStyle, but what the hell? ;-)
-        QWidgetList *list = QApplication::allWidgets();
-        QWidgetListIt it( *list );
-        QWidget *w;
-        while ((w=it.current()) != 0 ){
-            ++it;
-            if(w->inherits("QPushButton")){
-                ((LiquidStyle*)parent())->unapplyCustomAttributes((QPushButton *)w);
-            }
-        }
-        ((LiquidStyle*)parent())->loadCustomButtons();
-        it.toFirst();
-        while ((w=it.current()) != 0 ){
-            ++it;
-            if(w->inherits("QPushButton")){
-                ((LiquidStyle*)parent())->applyCustomAttributes((QPushButton *)w);
-            }
-        }
-
-    }
-}
 
 LiquidStyle::LiquidStyle()
-    :KStyle()
+    :QWindowsStyle()
 {
+    setName ( "LiquidStyle" );
+
     btnMaskBmp = QBitmap(37, 26, buttonmask_bits, true);
     btnMaskBmp.setMask(btnMaskBmp);
     htmlBtnMaskBmp = QBitmap(37, 26, htmlbuttonmask_bits, true);
@@ -730,20 +670,19 @@
     }
 
     loadCustomButtons();
-    lowLightVal = 100 + (2*KGlobalSettings::contrast()+4)*10;
+    lowLightVal = 100 + (2* /*KGlobalSettings::contrast()*/ 3 +4)*10;
     btnDict.clear();
     btnBorderDict.clear();
     bevelFillDict.clear();
     smallBevelFillDict.clear();
 
-    KConfig *config = KGlobal::config();
-    QString oldGrp = config->group();
+    Config config ( "qpe" );
     QPalette pal = QApplication::palette();
 
     // button color stuff
-    config->setGroup("General");
-    QColor c = config->readColorEntry("buttonBackground", &Qt::lightGray);
-    if(c == config->readColorEntry("background", &Qt::lightGray)){
+    config. setGroup ( "Appearance" );
+    QColor c = QColor ( config. readEntry("Button", ( Qt::lightGray ). name ( )));	
+    if ( c == QColor ( config. readEntry ( "background", ( Qt::lightGray ). name ( )))) {
         // force button color to be different from background
         QBrush btnBrush(QColor(200, 202, 228));
         appPal.setBrush(QColorGroup::Button, btnBrush);
@@ -794,15 +733,7 @@
     pagerBrush.setPixmap(*pix);
 
     // background color stuff
-    c = config->readColorEntry("background", &Qt::lightGray);
-    if(qstrcmp(kapp->argv()[0], "kicker") == 0){
-        appPal.setColor(QColorGroup::Mid, menuBrush.color().dark(110));
-        appPal.setColor(QColorGroup::Dark, menuBrush.color().dark(130));
-        appPal.setColor(QColorGroup::Midlight, menuBrush.color().light(110));
-        appPal.setColor(QColorGroup::Light, menuBrush.color().light(115));
-        menuBrush.setColor(c); // hack - used for kicker applets
-        appPal.setBrush(QColorGroup::Background, menuBrush);
-    }
+    c = QColor ( config. readEntry ( "Background", ( Qt::lightGray ).name ( )));
     c.hsv(&bH, &bS, &bV);
     c.light(120).hsv(&bHoverH, &bHoverS, &bHoverV);
 
@@ -825,13 +756,10 @@
     painter.end();
     bgBrush.setColor(c);
     bgBrush.setPixmap(wallPaper);
-    if(qstrcmp(kapp->argv()[0], "kicker") != 0 &&
-       qstrcmp(kapp->argv()[0], "ksplash") != 0){
         appPal.setBrush(QColorGroup::Background, bgBrush);
-    }
 
     // lineedits
-    c = config->readColorEntry("windowBackground", &Qt::white);
+    c = QColor ( config. readEntry("Base", ( Qt::white). name ( )));
     QPixmap basePix;
     basePix.resize(32, 32);
     basePix.fill(c.rgb());
@@ -857,51 +785,27 @@
         }
     }
 
-    config->setGroup(oldGrp);
 }
 
 void LiquidStyle::polish(QWidget *w)
 {
     if(w->inherits("QMenuBar")){
-        ((QFrame*)w)->setLineWidth(0);
-        w->setBackgroundMode(QWidget::NoBackground);
+        //((QFrame*)w)->setLineWidth(0);
+        w->setBackgroundMode(QWidget::PaletteBackground);
         return;
     }
     if(w->inherits("QPopupMenu"))
         w->setBackgroundMode(QWidget::NoBackground);
 
     if(w->isTopLevel()){
-        if(!w->inherits("QPopupMenu") &&
-           qstrcmp(kapp->argv()[0], "kicker") == 0){
-            qWarning("Got panel toplevel %s", w->className());
-            // force extensions and child panels to use proper palette.
-            if(w->inherits("Panel")){
-                qWarning("Setting panel palette");
-                w->setPalette(kapp->palette());
-            }
-            else{
-                // reset palette for everything else
-                QPalette pal = kapp->palette();
-                pal.setBrush(QColorGroup::Background, bgBrush);
-                pal.setColor(QColorGroup::Mid, bgBrush.color().dark(130));
-                pal.setColor(QColorGroup::Dark, bgBrush.color().dark(150));
-                pal.setColor(QColorGroup::Midlight, bgBrush.color().light(110));
-                pal.setColor(QColorGroup::Light, bgBrush.color().light(120));
-                w->setPalette(pal);
-            }
-        }
         return;
     }
-    if(qstrcmp(kapp->argv()[0], "kicker") == 0 &&
-       w->inherits("KPanelExtension")){ // FIXME - doesn't work
-        w->setPalette(kapp->palette());
-    }
+    
+    w-> setBackgroundOrigin ( QWidget::ParentOrigin );
 
     if(w->inherits("QComboBox") ||
        w->inherits("QLineEdit") || w->inherits("QRadioButton") ||
-       w->inherits("QCheckBox") || w->inherits("QScrollBar") ||
-       w->isA("AppletHandle") || w->inherits("KMiniPagerButton") ||
-       w->inherits("TaskContainer")){
+       w->inherits("QCheckBox") || w->inherits("QScrollBar")) {
         w->installEventFilter(this);
     }
     if(w->inherits("QLineEdit")){
@@ -913,6 +817,9 @@
         applyCustomAttributes((QPushButton *)w);
         w->installEventFilter(this);
     }
+    if(w->inherits("QButton") || w-> inherits("QComboBox")){
+    	w-> setBackgroundMode ( QWidget::PaletteBackground );
+    }
 
     bool isViewport = qstrcmp(w->name(), "qt_viewport") == 0 ||
         qstrcmp(w->name(), "qt_clipped_viewport") == 0;
@@ -942,11 +849,12 @@
         w->setMouseTracking(true);
         w->installEventFilter(this);
     }
+    if(w-> inherits("QToolButton")) {
+    	((QToolButton*)w)->setAutoRaise (false);
+    }
     if(w->ownPalette() && !w->inherits("QButton") && !w->inherits("QComboBox")){
         return;
     }
-    if(w->inherits("PanelButtonBase"))
-        return;
 
     if(w->parent() && w->parent()->isWidgetType() && !((QWidget*)w->parent())->
        palette().active().brush(QColorGroup::Background).pixmap()){
@@ -954,17 +862,18 @@
         return;
     }
     if(!isViewport && !isViewportChild && !w->testWFlags(WType_Popup) &&
-       !w->inherits("KDesktop") && !w->inherits("PasswordDlg")){
+	   !( !w-> inherits("QLineEdit") && w-> parent() && w-> parent()-> isWidgetType ( ) && w-> parent()-> inherits ( "QMultiLineEdit" ))) {
         if(w->backgroundMode() == QWidget::PaletteBackground ||
            w->backgroundMode() == QWidget::PaletteButton){
             w->setBackgroundMode(QWidget::X11ParentRelative);
         }
     }
-    if(w->inherits("KToolBar")){
+    if(w->inherits("QToolBar")){
         w->installEventFilter(this);
-        //w->setBackgroundMode(QWidget::NoBackground);
+        w->setBackgroundMode(QWidget::PaletteBackground);
         return;
     }
+
 }
 
 void LiquidStyle::unPolish(QWidget *w)
@@ -1001,12 +910,14 @@
         unapplyCustomAttributes((QPushButton *)w);
         w->removeEventFilter(this);
     }
-
+/*
+    if(w->inherits("QPushButton") || w-> inherits("QComboBox")){
+    	w-> setBackgroundMode ( PaletteBackground );
+    }
+*/
     if(w->inherits("QComboBox") ||
        w->inherits("QLineEdit") || w->inherits("QRadioButton") ||
-       w->inherits("QCheckBox") || w->inherits("QScrollBar") ||
-       w->isA("AppletHandle") || w->inherits("KMiniPagerButton") ||
-       w->inherits("TaskContainer")){
+       w->inherits("QCheckBox") || w->inherits("QScrollBar")) {
         w->removeEventFilter(this);
     }
     if(w->inherits("QButton") || w->inherits("QComboBox")){
@@ -1014,9 +925,9 @@
             w->setAutoMask(false);
         }
     }
-    if(w->inherits("KToolBar")){
+    if(w->inherits("QToolBar")){
         w->removeEventFilter(this);
-        //w->setBackgroundMode(QWidget::PaletteBackground);
+        w->setBackgroundMode(QWidget::PaletteBackground);
         return;
     }
     if(w->inherits("QHeader")){
@@ -1028,20 +939,24 @@
 void LiquidStyle::polish(QApplication *app)
 {
 
-    KStyle::polish(app);
+    QWindowsStyle::polish(app);
     menuAni = app->isEffectEnabled(UI_AnimateMenu);
     menuFade = app->isEffectEnabled(UI_FadeMenu);
     if(menuAni)
         app->setEffectEnabled(UI_AnimateMenu, false);
     if(menuFade)
         app->setEffectEnabled(UI_FadeMenu, false);
+	
+    qt_set_draw_menu_bar_impl((QDrawMenuBarItemImpl) &LiquidStyle::drawMenuBarItem);
 }
 
 void LiquidStyle::unPolish(QApplication *app)
 {
-    KStyle::unPolish(app);
+    QWindowsStyle::unPolish(app);
     app->setEffectEnabled(UI_AnimateMenu, menuAni);
     app->setEffectEnabled(UI_FadeMenu, menuFade);
+
+    qt_set_draw_menu_bar_impl ( 0 );
 }
 
 /*
@@ -1063,7 +978,7 @@
  */
 bool LiquidStyle::eventFilter(QObject *obj, QEvent *ev)
 {
-    if(obj->inherits("KToolBar")){
+    if(obj->inherits("QToolBar")){
         if(ev->type() == QEvent::Resize){
             const QObjectList *tbChildList = obj->children();
             QObjectListIt it(*tbChildList);
@@ -1076,35 +991,7 @@
 
         }
     }
-    else if(obj->inherits("KMiniPagerButton")){
-        QButton *btn = (QButton *)obj;
-        if(ev->type() == QEvent::Paint){
-            if(!(btn->isOn() || btn->isDown())){
-                QPalette pal = btn->palette();
-                pal.setBrush(QColorGroup::Dark, btn == highlightWidget ?
-                             pagerHoverBrush : pagerBrush);
-                btn->setPalette(pal);
-            }
-            else{
-                QPalette pal = btn->palette();
-                pal.setBrush(QColorGroup::Dark,
-                             QApplication::palette().active().brush(QColorGroup::Dark));
-                btn->setPalette(pal);
-
-            }
-        }
-        else if(ev->type() == QEvent::Enter){
-            highlightWidget = btn;
-            btn->repaint(false);
-        }
-        else if(ev->type() == QEvent::Leave){
-            highlightWidget = NULL;
-            btn->repaint(false);
-        }
-
-    }
-    else if(obj->inherits("QPushButton") || obj->inherits("QComboBox") ||
-            obj->isA("AppletHandle")){
+    else if(obj->inherits("QPushButton") || obj->inherits("QComboBox")){
         QWidget *btn = (QWidget *)obj;
         if(ev->type() == QEvent::Enter){
             if(btn->isEnabled()){
@@ -1119,20 +1006,7 @@
             }
         }
     }
-    else if(obj->inherits("TaskContainer")){
-        QButton *btn = (QButton *)obj;
-        QPalette pal = btn->palette();
-        if(ev->type() == QEvent::Enter){
-            pal.setColor(QColorGroup::Background, pal.active().button().light(110));
-            btn->setPalette(pal);
-        }
-        else if(ev->type() == QEvent::Leave){
-            pal.setColor(QColorGroup::Background,
-                         QApplication::palette().active().background());
-            btn->setPalette(pal);
-        }
-    }
-    else if(obj->inherits("QToolButton") && !obj->inherits("KToolBarButton")){
+    else if(obj->inherits("QToolButton")){
         QToolButton *btn = (QToolButton *)btn;
         if(!btn->autoRaise()){
             if(btn->isEnabled()){
@@ -1340,11 +1214,6 @@
     QColorGroup g = btn->colorGroup();
 
 
-    QColor testColor;
-    if(btn->parent() && btn->parent()->isWidgetType()){
-        testColor = p->backgroundColor(); // remove me
-    }
-
     //int dw = buttonDefaultIndicatorWidth();
     if(btn->hasFocus() || btn->isDefault()){
         QColor c = btn->hasFocus() ? g.button().light(110) : g.background();
@@ -2065,203 +1934,24 @@
     p->drawLineSegments(a);
 }
 
-void LiquidStyle::drawKBarHandle(QPainter *p, int x, int y, int w, int h,
-                             const QColorGroup &g, KToolBarPos,
-                             QBrush *)
-{
-    p->setPen(g.button().dark(120));
-    int x2 = x+w-1;
-    int y2 = y+h-1;
-    p->drawLine(x+1, y, x2-1, y);
-    p->drawLine(x+1, y2, x2-1, y2);
-    p->drawLine(x, y+1, x, y2-1);
-    p->drawLine(x2, y+1, x2, y2-1);
-
-    p->setPen(g.background());
-    p->drawPoint(x, y);
-    p->drawPoint(x2, y);
-    p->drawPoint(x, y2);
-    p->drawPoint(x2, y2);
-
-
-
-   // p->drawRect(x, y, w, h);
-    QPixmap *pix = bevelFillDict.find(g.button().rgb());
-    if(!pix){
-        int h, s, v;
-        g.button().hsv(&h, &s, &v);
-        pix = new QPixmap(*bevelFillPix);
-        adjustHSV(*pix, h, s, v);
-        bevelFillDict.insert(g.button().rgb(), pix);
-    }
-
-    p->drawTiledPixmap(x+1, y+1, w-2, h-2, *pix);
-}
-
-void LiquidStyle::drawKMenuBar(QPainter *p, int x, int y, int w, int h,
-                           const QColorGroup &g, bool mac, QBrush *)
-{
-    if(p->device() && p->device()->devType() == QInternal::Widget &&
-       ((KMenuBar *)p->device())->isTopLevelMenu()){
-        p->setPen(Qt::black);
-        p->drawRect(x, y, w, h);
-        p->drawTiledPixmap(x+1, y+1, w-2, h-2, *menuPix);
-        // left
-        p->drawLine(x+1, y+1, x+1, y+5);
-        p->drawLine(x+2, y+1, x+2, y+3);
-        p->drawLine(x+3, y+1, x+3, y+2);
-        p->drawLine(x+4, y+1, x+6, y+1);
-        // right
-        int x2 = x+w-1;
-        p->drawLine(x2-1, y+1, x2-1, y+5);
-        p->drawLine(x2-2, y+1, x2-2, y+3);
-        p->drawLine(x2-3, y+1, x2-3, y+2);
-        p->drawLine(x2-4, y+1, x2-6, y+1);
-    }
-    else{
-        qDrawShadePanel(p, x, y, w, h, g, false, 1,
-                        &g.brush(QColorGroup::Background));
-    }
 
-}
-
-void LiquidStyle::drawKToolBar(QPainter *p, int x, int y, int w, int h,
-                           const QColorGroup &g, KToolBarPos, QBrush *)
+void LiquidStyle::drawMenuBarItem(QPainter *p, int x, int y, int w, int h,
+                            QMenuItem *mi, QColorGroup &g, bool enabled, bool active )
 {
-    //p->fillRect(x, y, w, h, g.brush(QColorGroup::Background));
-}
+    x -= 2; // Bug in Qt/E 
+    y -= 2;
+    w += 2;
+    h += 2;
 
-void LiquidStyle::drawKToolBarButton(QPainter *p, int x, int y, int w, int h,
-                                    const QColorGroup &g, bool sunken,
-                                    bool raised, bool enabled, bool popup,
-                                    KToolButtonType icontext,
-                                    const QString& btext, const QPixmap *pixmap,
-                                    QFont *font, QWidget *btn)
-{
-    int dx, dy;
-
-    QFontMetrics fm(*font);
-
-    QToolBar* toolbar = 0;
-    if(btn->parent() && btn->parent()->isWidgetType() && btn->parent()->inherits("QToolBar"))
-        toolbar = static_cast<QToolBar*>(btn->parent());
-
-    --w, --h;
-    if(sunken)
-        ++x, ++y;
+    p-> fillRect ( x, y, w, h, g.brush(QColorGroup::Background));
 
-    QColor btnColor(sunken ?  g.button() : raised ? g.button().light(110) :
-                    g.background());
-    drawClearBevel(p, x, y, w, h, btnColor, g.background());
-
-    p->setPen(g.text());
-
-    if (icontext == Icon){ // icon only
-        if (pixmap){
-            dx = ( w - pixmap->width() ) / 2;
-            dy = ( h - pixmap->height() ) / 2;
-            if ( sunken )
-            {
-                ++dx;
-                ++dy;
-            }
-            p->drawPixmap( x+dx, y+dy, *pixmap );
-        }
-    }
-    else if (icontext == IconTextRight){ // icon and text (if any)
-        if (pixmap){
-            dx = 4;
-            dy = ( h - pixmap->height() ) / 2;
-            if ( sunken ){
-                ++dx;
-                ++dy;
-            }
-            p->drawPixmap( x+dx, y+dy, *pixmap );
-        }
-        if (!btext.isNull()){
-            int tf = AlignVCenter|AlignLeft;
-            if (pixmap)
-                dx= 4 + pixmap->width() + 2;
-            else
-                dx= 4;
-            dy = 0;
-            if ( sunken ){
-                ++dx;
-                ++dy;
-            }
-            if (font)
-                p->setFont(*font);
-            if(raised)
-                p->setPen(KGlobalSettings::toolBarHighlightColor());
-            p->drawText(x+dx, y+dy, w-dx, h, tf, btext);
-        }
-    }
-    else if (icontext == Text){ // only text, even if there is a icon
-        if (!btext.isNull()){
-            int tf = AlignVCenter|AlignLeft;
-            if (!enabled)
-                p->setPen(g.dark());
-            dx= (w - fm.width(btext)) / 2;
-            dy= (h - fm.lineSpacing()) / 2;
-            if ( sunken ){
-                ++dx;
-                ++dy;
-            }
-            if (font)
-                p->setFont(*font);
-            if(raised)
-                p->setPen(KGlobalSettings::toolBarHighlightColor());
-            p->drawText(x+dx, y+dy, fm.width(btext), fm.lineSpacing(), tf, btext);
-        }
-    }
-    else if (icontext == IconTextBottom){
-        if (pixmap){
-            dx = (w - pixmap->width()) / 2;
-            dy = (h - fm.lineSpacing() - pixmap->height()) / 2;
-            if ( sunken ){
-                ++dx;
-                ++dy;
-            }
-            p->drawPixmap( x+dx, y+dy, *pixmap );
-        }
-        if (!btext.isNull()){
-            int tf = AlignBottom|AlignHCenter;
-            dy= pixmap->height();
-            dx = 2;
-            if ( sunken ){
-                ++dx;
-                ++dy;
-            }
-            if (font)
-                p->setFont(*font);
-            if(raised)
-                p->setPen(KGlobalSettings::toolBarHighlightColor());
-            p->drawText(x, y, w, h-3, tf, btext);
-        }
-    }
-    if (popup){
-        if (enabled)
-            qDrawArrow (p, Qt::DownArrow, Qt::WindowsStyle, false, w-5, h-5, 0, 0,
-                        g, true);
-        else
-            qDrawArrow (p, Qt::DownArrow, Qt::WindowsStyle, false, w-5, h-5,
-                        0, 0, g, false);
-    }
-}
-
-
-void LiquidStyle::drawKMenuItem(QPainter *p, int x, int y, int w, int h,
-                            const QColorGroup &g, bool active, QMenuItem *mi,
-                            QBrush *)
-{
-    if ( p->font() == KGlobalSettings::generalFont() )
-        p->setFont( KGlobalSettings::menuFont() );
+   w -= 2;
 
     if(menuHandler->useShadowText()){
         QColor shadow;
         if(p->device() && p->device()->devType() == QInternal::Widget &&
-           ((QWidget *)p->device())->inherits("KMenuBar")){
-            shadow = ((KMenuBar*)p->device())->isTopLevel() ? g.button().dark(130) :
+           ((QWidget *)p->device())->inherits("QMenuBar")){
+            shadow = ((QMenuBar*)p->device())->isTopLevel() ? g.button().dark(130) :
                 g.background().dark(130);
         }
         else
@@ -2336,8 +2026,6 @@
 
     maxpmw = QMAX( maxpmw, 20 );
 
-    if ( p->font() == KGlobalSettings::generalFont() )
-        p->setFont( KGlobalSettings::menuFont() );
 
     bool dis = !enabled;
     QColorGroup itemg = dis ? pal.disabled() : pal.active();
@@ -2363,7 +2051,7 @@
             p->fillRect(x, y, w, h, menuBrush);
         }
         else{
-            KPixmap *pix = menuHandler->pixmap(((QWidget*)p->device())->winId());
+            QPixmap *pix = menuHandler->pixmap(((QWidget*)p->device())->winId());
             if(pix)
                 p->drawPixmap(x, y, *pix, x, y, w, h);
         }
@@ -2508,25 +2196,6 @@
     return h;
 }
 
-void LiquidStyle::drawKProgressBlock(QPainter *p, int x, int y, int w, int h,
-                                    const QColorGroup &g, QBrush *fill)
-{
-    p->setPen(g.button().dark(130));
-    p->drawRect(x, y, w, h);
-    p->setPen(g.button().light(120));
-    p->drawRect(x+1, y+1, w-2, h-2);
-    if(w >= 4 && h >= 4){
-        QPixmap *pix = bevelFillDict.find(g.button().dark(120).rgb());
-        if(!pix){
-            int h, s, v;
-            g.button().dark(120).hsv(&h, &s, &v);
-            pix = new QPixmap(*bevelFillPix);
-            adjustHSV(*pix, h, s, v);
-            bevelFillDict.insert(g.button().dark(120).rgb(), pix);
-        }
-        p->drawTiledPixmap(x+2, y+2, w-4, h-4, *pix);
-    }
-}
 
 void LiquidStyle::drawFocusRect(QPainter *p, const QRect &r,
                             const QColorGroup &g, const QColor *c,
@@ -2540,25 +2209,25 @@
             return;
         }
         else{
-            KStyle::drawFocusRect(p, r, g, c, atBorder);
+            QWindowsStyle::drawFocusRect(p, r, g, c, atBorder);
         }
     }
     else
-        KStyle::drawFocusRect(p, r, g, c, atBorder);
+        QWindowsStyle::drawFocusRect(p, r, g, c, atBorder);
 
 }
 
 void LiquidStyle::polishPopupMenu(QPopupMenu *mnu)
 {
     mnu->installEventFilter(menuHandler);
-    KStyle::polishPopupMenu(mnu);
+    QWindowsStyle::polishPopupMenu(mnu);
 }
 
 void LiquidStyle::drawTab(QPainter *p, const QTabBar *tabBar, QTab *tab,
                       bool selected)
 {
     if(tabBar->shape() != QTabBar::RoundedAbove){
-        KStyle::drawTab(p, tabBar, tab, selected);
+        QWindowsStyle::drawTab(p, tabBar, tab, selected);
         return;
     }
     QPixmap tilePix;
@@ -2671,7 +2340,7 @@
         vFrame = 8; // was 10
     }
     else
-        KStyle::tabbarMetrics(t, hFrame, vFrame, overlap);
+        QWindowsStyle::tabbarMetrics(t, hFrame, vFrame, overlap);
 }
 
 
@@ -2699,7 +2368,7 @@
         p->drawLine(x+1, y+1, x+1, y2-1);
     }
     else if(lineWidth != 2 || !sunken)
-        KStyle::drawPanel(p, x, y, w, h, g, sunken, lineWidth, fill);
+        QWindowsStyle::drawPanel(p, x, y, w, h, g, sunken, lineWidth, fill);
     else{
         QPen oldPen = p->pen();
         int x2 = x+w-1;
@@ -2726,105 +2395,6 @@
     }
 }
 
-void LiquidStyle::drawKickerAppletHandle(QPainter *p, int x, int y, int w, int h,
-                                      const QColorGroup &g, QBrush *)
-{
-    p->fillRect(x, y, w, h, g.brush(QColorGroup::Background));
-    drawClearBevel(p, x, y, w, h, highlightWidget == p->device() ?
-                   g.button().light(120) : g.button(), g.button());
-    /*
-    if(h > w){
-        int y2 = y+h-1;
-
-        p->setPen(g.light());
-
-        p->drawLine(x+1, y+2, x+1, y2-2);
-        p->drawLine(x+4, y+2, x+4, y2-2);
-
-        p->setPen(g.dark());
-        p->drawLine(x+2, y+2, x+2, y2-2);
-        p->drawLine(x+5, y+2, x+5, y2-2);
-
-    }
-    else{
-        int x2 = x+w-1;
-
-        p->setPen(g.light());
-
-        p->drawLine(x+2, y+1, x2-2, y+1);
-        p->drawLine(x+2, y+4, x2-2, y+4);
-
-        p->setPen(g.dark());
-        p->drawLine(x+2, y+2, x2-2, y+2);
-        p->drawLine(x+2, y+5, x2-2, y+5);
-    }*/
-
-}
-
-void LiquidStyle::drawKickerTaskButton(QPainter *p, int x, int y, int w, int h,
-                                  const QColorGroup &g,
-                                  const QString &text, bool sunken,
-                                  QPixmap *pixmap, QBrush *)
-{
-    p->fillRect(x, y, w, h, g.brush(QColorGroup::Button));
-    drawClearBevel(p, x, y, w, h, sunken ? g.button() : g.background(), g.button());
-    p->setPen(g.buttonText()); // Kicker doesn't set this ;-)
-
-    if(text.isNull() && !pixmap)
-        return;
-
-    const int pxWidth = 20;
-    int textPos = pxWidth;
-    QRect br(buttonRect(x, y, w, h));
-
-    if (sunken)
-        p->translate(1,1);
-
-    if ( pixmap && !pixmap->isNull() ) {
-        int dx = ( pxWidth - pixmap->width() ) / 2;
-        int dy = ( h - pixmap->height() ) / 2;
-        p->drawPixmap( br.x()+dx, dy, *pixmap );
-    }
-
-    QString s = text;
-    static const QString &modStr = KGlobal::staticQString(
-           QString::fromUtf8("[") + i18n("modified") + QString::fromUtf8("]"));
-
-    int modStrPos = s.find(modStr);
-
-    if (-1 != modStrPos) {
-
-      // +1 because we include a space after the closing brace.
-      s.remove(modStrPos, modStr.length()+1);
-
-      QPixmap modPixmap = SmallIcon("modified");
-
-      int dx = (pxWidth   - modPixmap.width())  / 2;
-      int dy = (h  - modPixmap.height()) / 2;
-
-      p->drawPixmap(br.x() + textPos + dx, dy, modPixmap);
-
-      textPos += pxWidth;
-    }
-
-    if (!s.isEmpty()){
-        if (p->fontMetrics().width(s) > br.width() - textPos) {
-
-            int maxLen = br.width() - textPos - p->fontMetrics().width("...");
-
-            while ((!s.isEmpty()) && (p->fontMetrics().width(s) > maxLen))
-                s.truncate(s.length() - 1);
-
-            s.append("...");
-        }
-
-        p->setPen(g.buttonText());
-
-        p->drawText(br.x()+ textPos, -1, w-textPos, h,
-                    AlignLeft|AlignVCenter, s);
-    }
-
-}
 
 void LiquidStyle::adjustHSV(QPixmap &pix, int h, int s, int v)
 {
@@ -2998,22 +2568,22 @@
     customBtnIconList.clear();
     customBtnLabelList.clear();
 
-    KConfig *config = KGlobal::config();
-    QString oldGrp = config->group();
-    config->setGroup("MosfetButtons");
+//    KConfig *config = KGlobal::config();
+//    QString oldGrp = config->group();
+//    config->setGroup("MosfetButtons");
 
     QStrList iconList, colorList; //temp, we store QPixmaps and QColors
     iconList.setAutoDelete(true);
     colorList.setAutoDelete(true);
-    config->readListEntry("Labels", customBtnLabelList);
-    config->readListEntry("Icons", iconList);
-    config->readListEntry("Colors", colorList);
+//    config->readListEntry("Labels", customBtnLabelList);
+//    config->readListEntry("Icons", iconList);
+//    config->readListEntry("Colors", colorList);
 
     const char *labelStr = customBtnLabelList.first();
     const char *colorStr = colorList.first();
     const char *iconStr = iconList.first();
 
-    KIconLoader *ldr = KGlobal::iconLoader();
+//    KIconLoader *ldr = KGlobal::iconLoader();
     while(labelStr != NULL){
         QColor *c = new QColor;
         c->setNamedColor(QString(colorStr));
@@ -3022,7 +2592,7 @@
         QString tmpStr(iconStr);
         if(!tmpStr.isEmpty()){
             QPixmap *pixmap =
-                new QPixmap(ldr->loadIcon(tmpStr, KIcon::Small));
+                new QPixmap();//ldr->loadIcon(tmpStr, KIcon::Small));
             if(pixmap->isNull()){
                 delete pixmap;
                 customBtnIconList.append(NULL);
@@ -3037,7 +2607,6 @@
         colorStr = colorList.next();
         iconStr = iconList.next();
     }
-    config->setGroup(oldGrp);
 }
 
 void LiquidStyle::applyCustomAttributes(QPushButton *btn)
@@ -3087,7 +2656,7 @@
     }
 }
 
-#include "liquid.moc"
+// #include "liquid.moc"
 
 
 
--- -	Mon Jun 24 19:05:49 2002
+++ plugin.cpp	Thu Jun 20 20:01:37 2002
@@ -1,29 +1,29 @@
 #include "liquid.h"
-#include <klocale.h>
+
 
 extern "C" {
-    KStyle* allocate();
-    int minor_version();
-    int major_version();
-    const char *description();
+    QStyle* allocate ( );
+    int minor_version ( );
+    int major_version ( );
+    const char *description ( );
 }
 
-KStyle* allocate()
+QStyle* allocate ( )
 {
-    return(new LiquidStyle);
+    return new LiquidStyle ( );
 }
 
-int minor_version()
+int minor_version ( )
 {
-    return(0);
+    return 0;
 }
 
-int major_version()
+int major_version ( )
 {
-    return(1);
+    return 1;
 }
 
-const char *description()
+const char *description ( )
 {
-    return(i18n("High performance liquid plugin").utf8());
+    return "High performance liquid plugin";
 }
